domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty.go

```diff
@@ -111,18 +112,21 @@ func (v *blockValidator) checkParentHeadersExist(header externalapi.BlockHeader)
 			return err
 		}
 		if !parentHeaderExists {
+			parentStatus, err := v.blockStatusStore.Get(v.databaseContext, parent)
+			isNotFoundError := database.IsNotFoundError(err)
+			if !isNotFoundError {
+				if err != nil {
```

Comment:
Please check that an error exists first and whether it's a not-found error second.